sql/6_0_0-to-6_1_0_upgrade.sql

```diff
@@ -199,3 +199,30 @@ ALTER TABLE `procedure_providers` ADD `type` VARCHAR(31) DEFAULT NULL;
 ALTER TABLE `procedure_answers` ADD `procedure_code` VARCHAR(31) DEFAULT NULL;
 #EndIf
 
+#IfNotRow users username oe-system
+INSERT INTO `users`(`username`,`password`,`lname`,`authorized`,`active`) VALUES ('oe-system','NoLogin','System Operation User',0,0);
+INSERT INTO `gacl_aro`(`id`, `section_value`, `value`, `order_value`, `name`, `hidden`)
+    SELECT max(`id`)+1,'users','oe-system',10,'System Operation User', 0 FROM `gacl_aro`;
+INSERT INTO `gacl_groups_aro_map`(`group_id`, `aro_id`)
+    VALUES (
+        (SELECT `id` FROM `gacl_aro_groups` WHERE parent_id=10 AND value='admin')
+        ,(SELECT `id` FROM `gacl_aro` WHERE `section_value` = 'users' AND `value` = 'oe-system')
+    );
+#EndIf
+
+#IfNotTable
+CREATE TABLE `export_job` (
+  `uuid` varchar(40) NOT NULL,
+  `user_id` varchar(40) NOT NULL,
+  `client_id` varchar(80) NOT NULL,
+  `status` varchar(40) NOT NULL,
+  `start_time` datetime NOT NULL,
```

Comment:
datetime should default to NULL

Reply:
This is a required field, why would we default to NULL?

Reply:
This is a required field, why would we default to NULL?

Reply:
project is in process of supporting sql standards. one of them is the datetime/timestamp fields need to either be a valid date or null (no blank or zero dates). more details here:
https://github.com/openemr/openemr/issues/3378

Reply:
btw, meant to say mysql strict mode rather than sql standards above

Reply:
TODO for me. revisit/review why recommending null for empty date

Reply:
https://community.open-emr.org/t/value-0000-00-00-is-not-valid-for-mysql-data-type-date-bug-for-mysql-5-5-49-win32/7978/6?u=stephenwaite

Reply:
thanks for digging up that thread @stephenwaite .

Did a quick test to see what happens when insert columns with nothing in it (except for the `nothing interesting` field where inserted `hey) on the following
![image](https://user-images.githubusercontent.com/278968/106860630-81d18500-6679-11eb-9f20-824511d806ad.png)

The ability to insert 0000-00-00 is definitely problematic since it then needs special care as we convert to a strict mode compatible database. It means in upgrade script we need to change field to allow null, disable strict mode, convert the 0000-00-00, then enable strict mode (note there are examples where we did this in the 6.0.0 upgrade script since a student during gsoc worked on this a bit.

So, for datetime/date/timestamp, we need to keep them at default NULL (ie. can not use NOT NULL).